refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsEmptyTest.java

```diff
@@ -0,0 +1,331 @@
+            "  }",
+            "",
+            "  int[][] negative4() {",
+            "    return new int[][] {{0}};",
+            "  }",
+            "",
+            "  List<Integer> negative5() {",
+            "    return new ArrayList<>(ImmutableList.of(1));",
+            "  }",
+            "",
+            "  Map<Integer, Integer> negative6() {",
+            "    return new HashMap<>(ImmutableMap.of(1, 2));",
+            "  }",
+            "",
+            "  Map<Integer, Integer> negative7() {",
+            "    return new LinkedHashMap<>(ImmutableMap.of(1, 2));",
+            "  }",
+            "",
+            "  ImmutableList<Integer> negative8() {",
+            "    return ImmutableList.of(1);",
+            "  }",
+            "",
+            "  ImmutableSet<Integer> negative9() {",
+            "    return ImmutableSet.of(1);",
+            "  }",
+            "",
+            "  ImmutableMap<Integer, Integer> negative10() {",
+            "    return ImmutableMap.of(1, 2);",
+            "  }",
+            "",
+            "  ImmutableSetMultimap<Integer, Integer> negative11() {",
+            "    return ImmutableSetMultimap.of(1, 2);",
+            "  }",
+            "",
+            "  List<Integer> negative12() {",
+            "    return List.of(1);",
+            "  }",
+            "",
+            "  Map<Integer, Integer> negative13() {",
+            "    return Map.of(1, 2);",
+            "  }",
+            "",
+            "  Set<Integer> negative14() {",
+            "    return Set.of(1);",
+            "  }",
+            "",
+            "  Stream<Integer> negative15() {",
+            "    return Stream.of(1);",
+            "  }",
+            "",
+            "  int[] positive1() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new int[0];",
+            "  }",
+            "",
+            "  int[][] positive2() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new int[0][1];",
+            "  }",
+            "",
+            "  int[] positive3() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new int[] {};",
+            "  }",
+            "",
+            "  int[][] positive4() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new int[][] {};",
+            "  }",
+            "",
+            "  List<Integer> positive5() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new ArrayList<>();",
+            "  }",
+            "",
+            "  List<String> positive6() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new ArrayList<>(1);",
+            "  }",
+            "",
+            "  List<String> positive7() {",
+            "    // BUG: Diagnostic contains:",
+            "    return new ArrayList<>(",
+            "        // BUG: Diagnostic contains:",
+            "        ImmutableList.of());",
```

Comment:
Here and below: expressions split across multiple lines, to prove that the outer expression is (also) recognized as an empty collection.